Formula/vis.rb

```diff
@@ -40,19 +42,30 @@ def install
 
     env = { LUA_PATH: ENV["LUA_PATH"], LUA_CPATH: ENV["LUA_CPATH"] }
     bin.env_script_all_files(libexec/"bin", env)
-    # Rename vis & the matching manpage to avoid clashing with the system.
-    mv bin/"vis", bin/"vise"
-    mv man1/"vis.1", man1/"vise.1"
+
+    on_macos do
+      # Rename vis & the matching manpage to avoid clashing with the system.
+      mv bin/"vis", bin/"vise"
+      mv man1/"vis.1", man1/"vise.1"
+    end
   end
 
   def caveats
-    <<~EOS
-      To avoid a name conflict with the macOS system utility /usr/bin/vis,
-      this text editor must be invoked by calling `vise` ("vis-editor").
-    EOS
+    on_macos do
+      <<~EOS
+        To avoid a name conflict with the macOS system utility /usr/bin/vis,
+        this text editor must be invoked by calling `vise` ("vis-editor").
+      EOS
+    end
   end
 
   test do
-    assert_match "vis v#{version} +curses +lua", shell_output("#{bin}/vise -v 2>&1")
+    binary = bin/"vise"
+
+    on_linux do
+      binary = bin/"vis"
+    end
```

Comment:
Pushed a minor fix for the test since this PR modifies the formula to rename `vis` to `vise` only on macOS.

Reply:
Still fails though, sadly.